homeassistant/components/motionmount/coordinator.py

```diff
@@ -0,0 +1,45 @@
+"""Update coordinator for the MotionMount."""
+import logging
+from typing import Any
+
+import motionmount  # type: ignore[import-untyped]
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class MotionMountCoordinator(DataUpdateCoordinator):
+    """Coordinator for MotionMount."""
+
+    def __init__(self, hass: HomeAssistant) -> None:
+        """Initialize the MotionMount coordinator."""
+        super().__init__(hass, _LOGGER, name="MotionMount")
+        self._mm: motionmount.MotionMount = None
+
+    @property
+    def mm(self) -> motionmount.MotionMount:
+        """Returns the MotionMount."""
+        return self._mm
+
+    @mm.setter
+    def mm(self, new_value: motionmount.MotionMount) -> None:
+        self._mm = new_value
+
+    async def _async_update_data(self) -> dict[str, Any]:
+        return self._get_data_from_motionmount()
+
+    def motionmount_callback(self) -> None:
+        """Update data from updated MotionMount."""
+        self.async_set_updated_data(self._get_data_from_motionmount())
```

Comment:
For the callback, can't we instead ask the coordinator to do a refresh?

Reply:
I'm sorry, but the exact working of the `DataUpdateCoordinator` is a bit mysterious to me. You mean I call `await self.async_request_refresh()` in this callback and merge `_get_data_from_motionmount()` into `_async_update_data()`?

Reply:
Hi @joostlek, I tried to get such a thing working, but since the callback is a synchronous function I can't call `await self.async_request_refresh()`. Due you have a suggestion here?